docs/rst/tutorials/micro/custom_transports/custom_transports.rst

```diff
@@ -0,0 +1,278 @@
+.. _tutorials_micro_custom_transports:
+
+Custom Transports
+=================
+
+.. contents:: Table of Contents
+    :depth: 2
+    :local:
+    :backlinks: none
+
+
+This tutorial aims at providing step-by-step guidance for those users interested in creating micro-ROS custom transports, instead of using the ones provided by default in the micro-ROS tools set.
+
+This tutorial starts from a previously created micro-ROS environment. Check the first steps of `First micro-ROS application on an RTOS<../../core/first_application_rtos/>` for instructions on how to create a micro-ROS environment for embedded platforms.
+
+The micro-ROS middleware, *eProsima Micro XRCE-DDS*, provides a user API that allows interfacing with the lowest level transport layer at runtime, which enables users to implement their own transports in both the micro-ROS Client and micro-ROS Agent libraries.
+
+Thanks to this, the Micro XRCE-DDS wire protocol can be transmitted over virtually any protocol, network or communication mechanism. In order to do so, two general communication modes are provided:
+
+- **Stream-oriented mode**: the communication mechanism implemented does not have the concept of packet. `HDLC framing<https://micro-xrce-dds.docs.eprosima.com/en/latest/transport.html?highlight=hdlc#custom-serial-transport>` will be used.
+- **Packet-oriented mode**: the communication mechanism implemented is able to send a whole packet that includes an XRCE message.
+
+These two modes can be selected by activating and deactivating the ``framing`` parameter in both the micro-ROS Client and the micro-ROS Agent functions.
+
+micro-ROS Client
+----------------
+
+An example on how to set these external transport callbacks in the micro-ROS Client API is:
+
+.. code-block:: c
+
+    #include <rmw_microros/rmw_microros.h>
+
+    ...
+
+    struct custom_args {
+        ...
+    }
+
+    struct custom_args args;
+
+    rmw_uros_set_custom_transport(
+        true, // Framing enabled here. Using Stream-oriented mode.
```

Comment:
use the defines: https://github.com/micro-ROS/rmw_microxrcedds/blob/a2a74846971697e4afc8a70b59c9a8352d74195b/rmw_microxrcedds_c/include/rmw_microros/custom_transport.h#L30-L31